omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs2_0/EncounterSearchHandler2_0.java

```diff
@@ -73,7 +73,7 @@ public PageableResult search(RequestContext context) throws ResponseException {
 		EncounterType encounterType = ((EncounterTypeResource1_8) Context.getService(RestService.class)
 				.getResourceBySupportedClass(EncounterType.class)).getByUniqueId(encounterTypeUuid);
 
-		if (patient != null) {
+		if (patient != null && (encounterType != null || encounterTypeUuid == null)) {
```

Comment:
I don't really understand what this is doing with this `or` clause.  How is this invalid or unauthorized?

Reply:
Without the this additional clause, if search for a nonexistent `encounterType` or I lack privileges, the `encounterType` returned on line 73 will be null but the query will be done ignoring the explicitly passed `encounterType`.

e.g.:
If I call `/rest/v1/encounter?patient=<patient-UUID>&encounterType=123` (123 is a nonexistent encounter type). It will return me a list with all encounters for all other encounter types. I was expecting it to return an empty list.

@mseaton let me know if this makes sense or any further clarification is required.

Reply:
This is actually a more general principle we should be following: if the user explicitly requests filtering by a property and the value that they specify for that property doesn't match any objects, then the result would should be an empty set.

That said, @icrc-psousa the description both here and in the ticket are, I think, worded in a way that only makes sense if you understand that data filter is being used here. We should reword things to, e.g., "Return empty result if specified encounter type cannot be found".

Reply:
I've reworded the PR and ticket title according to the above feedback.
@ibacher @mseaton please review when you have some time. Thanks!